c2corg_ui/static/js/documentservice.js

```diff
@@ -124,17 +124,23 @@ app.Document.prototype.hasAssociation = function(type, id) {
  * @param {appx.SimpleSearchDocument} doc
  * @param {string=} doctype Optional doctype
  * @param {boolean=} setOutingTitle
+ * @param {?boolean} editing
  * @export
  */
 app.Document.prototype.pushToAssociations = function(doc, doctype,
-    setOutingTitle) {
+    setOutingTitle, editing) {
   var associations = this.document.associations;
   doctype = doctype || app.utils.getDoctype(doc['type']);
   setOutingTitle = typeof setOutingTitle !== 'undefined' ?
     setOutingTitle : false;
   doc['new'] = true;
   associations[doctype].push(doc);
 
+  // when creating a route, make the first associated wp a main one
+  if (editing && associations.waypoints.length === 1) {
```

Comment:
OK my bad then!
I agree with your choices.
(this is not a code review though, sorry)